### PR TITLE
Feature/convex actions

### PR DESCRIPTION
**Pull request by kjaredic**

### CODE REVIEW

**Review by majkic99 — COMMENTED** [review]
looks good, see about hardcoding address(this) in withdraw

File: contracts/actions/convex/ConvexWithdraw.sol
Comment (yippee-ki-yay) [reply contracts/actions/convex/ConvexWithdraw.sol:60]: Hardcode sounds like a better solution

File: contracts/actions/convex/helpers/ConvexHelper.sol
Comment (yippee-ki-yay) [inline contracts/actions/convex/helpers/ConvexHelper.sol:38]: All constants that can't change no need to add constructor and initialize it on the fly

File: contracts/actions/convex/ConvexClaim.sol
Comment (yippee-ki-yay) [inline contracts/actions/convex/ConvexClaim.sol:19]: Is this something that will change or can we hardcode it?

File: contracts/actions/convex/ConvexClaim.sol
Comment (yippee-ki-yay) [inline contracts/actions/convex/ConvexClaim.sol:51]: Action*

File: contracts/actions/convex/helpers/ConvexHelper.sol
Comment (yippee-ki-yay) [inline contracts/actions/convex/helpers/ConvexHelper.sol:109]: Why is this check needed are we expecting earnRewards to return 0x0 sometimes and why stop sending of other tokens after that?

File: contracts/actions/convex/helpers/ConvexHelper.sol
Comment (yippee-ki-yay) [inline contracts/actions/convex/helpers/ConvexHelper.sol:72]: Add comments how it might leave empty 0x0 add the end if there are crv/cvx addresses

File: contracts/actions/convex/helpers/ConvexHelper.sol
Comment (yippee-ki-yay) [inline contracts/actions/convex/helpers/ConvexHelper.sol:111]: Do we ever need to pull tokens from somewhere, isn't this function only used to send tokens from user proxy to _to` address?

File: contracts/actions/convex/helpers/ConvexHelper.sol
Comment (yippee-ki-yay) [inline contracts/actions/convex/helpers/ConvexHelper.sol:101]: Missing underscore for argument inputs

File: contracts/views/ConvexView.sol
Comment (yippee-ki-yay) [inline contracts/views/ConvexView.sol:6]: It's a view but i see nothing

**Review by yippee-ki-yay — COMMENTED** [review]
Few details needs to be work out before merge and defisaver-sdk  branch to be fixed.

File: contracts/actions/convex/ConvexClaim.sol
Comment (kjaredic) [reply contracts/actions/convex/ConvexClaim.sol:19]: rewardContract differs from pool to pool

File: contracts/actions/convex/helpers/ConvexHelper.sol
Comment (kjaredic) [reply contracts/actions/convex/helpers/ConvexHelper.sol:111]: you can claim for a different address (as long as you have reward token approvals to move them)
the deposit action also allows staking for other addresses so there may be use cases out there :thinking:

File: contracts/actions/convex/ConvexWithdraw.sol
Comment (kjaredic) [reply contracts/actions/convex/ConvexWithdraw.sol]: hardcoded it is then, although this could cause misuse if the docs aren't read

File: contracts/views/ConvexView.sol
Comment (kjaredic) [reply contracts/views/ConvexView.sol:6]: will update as soon as we sync with front-end

**Review by yippee-ki-yay — APPROVED** [review]
lgtm

**Pull request merged**
